src/block/EndPortalFrame.php

```diff
@@ -58,4 +63,18 @@ public function getLightLevel() : int{
 	protected function recalculateCollisionBoxes() : array{
 		return [AxisAlignedBB::one()->trim(Facing::UP, 3 / 16)];
 	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($this->eye || $item->getTypeId() !== ItemTypeIds::ENDER_EYE){
+			return false;
+		}
+		$world = $this->getPosition()->getWorld();
+		$world->setBlock($this->getPosition(), $this->setEye(true));
+		$world->addSound($this->getPosition(), new EndPortalFrameFillSound());
```

Comment:
We don't need (or want) to use getPosition here - a clone is neither wanted nor required